### ISSUE TITLE
gate io watch order for perpetual

### ISSUE THREAD

**Issue by pickmone** [ccxt/ccxt#11971]
Hi, 

I am testing the watch order function for gate, but it does not seem to respond to orders made for the futures (swap) market. I am using the v4 api. 

Wondering what is the solution for this?

thank you,

**Comment by pickmone** [comment]
also on a related note, are watch_order_book providing the right data if we set the option: defaultType to be 'swap'?

I am looking at the gateio.py, and i only see one ws url ''ws': 'wss://ws.gate.io/v4' but per gateio api doc the url should be
wss://fx-ws.gateio.ws/v4/ws/usdt for USDT contract and wss://fx-ws.gateio.ws/v4/ws/btc for BTC contracts. 
(https://www.gate.io/docs/developers/futures/ws/en/#gate-io-futures-websocket-v4),

**Comment by pickmone** [comment]
Thanks! I just had a look at the watch_order_book function for gate. ie. the url is still url = self.urls['api']['ws']. Not sure if I am missing something but the url there is still using the old url. 

Watch_order looks good!

**Comment by carlosmiei** [comment]
@pickmone Yes watchOrderBook was not updated yet but I intend to do it soon

**Comment by pickmone** [comment]
ah ok. thanks!
